src/routes/moment/[id]/+server.ts

```diff
@@ -1,13 +1,50 @@
 import { json, error } from '@sveltejs/kit';
-import { supabase } from '$lib/clients/supabaseClient'
+import fs from 'fs';
+import path from 'path';
+
+type DescriptionsCacheType = Record<number, string>;
+
+let descriptionsCache: DescriptionsCacheType | null = null;
+
+async function loadDescriptions(): Promise<DescriptionsCacheType | null> {
+    if (descriptionsCache) {
+        return descriptionsCache;
+    }
+
+    const descriptionsFilePath = path.resolve('static/data', 'descriptions.json');
+
+    return new Promise((resolve, reject) => {
+        fs.readFile(descriptionsFilePath, 'utf8', (err, fileContent) => {
+            if (err) {
+                console.error('Error reading descriptions file:', err);
+                reject(error(500, 'Error fetching moment text'));
+            } else {
+                descriptionsCache = JSON.parse(fileContent);
+                resolve(descriptionsCache);
+            }
+        });
+    });
+}
 
 export async function GET({ params }) {
     const { id } = params;
-    const { data, error: fetchError } = await supabase.from('moments').select('description').eq('short_id', parseInt(id, 10)).single()
 
-    if (fetchError) {
-        error(500, 'Error fetching moment text')
-    }
+    try {
+        const descriptions = await loadDescriptions();
```

Comment:
i’m not sure how svelte server does this but it would be ideal that the json does not need to be parsed and loaded every time a user clicks a description. Seems like a computation that could be memoized.

Reply:
The code already memoises this. `loadDescriptions` creates an in-memory cache that is populated on the first request after the server is booted. Each click on a description uses this cache. This means the data is not reloaded and re-parsed every time a user clicks a description.